lib/main.js

```diff
@@ -261,39 +259,15 @@ const Hydrogen = {
   },
 
   consumeStatusBar(statusBar: atom$StatusBar) {
-    const statusBarElement = document.createElement("div");
-    statusBarElement.classList.add("inline-block", "hydrogen");
-
-    statusBar.addLeftTile({
-      item: statusBarElement,
-      priority: 100
-    });
-
-    const onClick = this.showKernelCommands.bind(this);
-
-    reactFactory(
-      <StatusBar store={store} onClick={onClick} />,
-      statusBarElement
+    return services.statusBar.addStatusBar(
```

Comment:
Good question. The easiest way to fix that would be to use the `store.subscriptions` composite disposable which gets cleaned up on deactivate. So instead of returning the disposable, just add it to `store.subscriptions.add(new Disposable(() => statusBarTile.destroy()))` inside of `addStatusBar`

Reply:
This shouldn't be necessary according to how the service api works

Reply:
Please view the [Service API Docs](https://flight-manual.atom.io/behind-atom/sections/interacting-with-other-packages-via-services/). 

>These methods will be called any time a package is activated that provides their corresponding service. They will receive the service object as an argument. You will usually need to perform some kind of cleanup in the event that the package providing the service is deactivated. To do this, return a Disposable from your service-consuming method:

Reply:
Cool. I thought an new `CompositeDisposables` for consumed services would be better than `store.subscriptions` might be better, but it's just my opinion. We can go with `store.subscriptions` of course.

Reply:
Why would adding a new one be better though?

Reply:
Umm nvm. I felt like that since I'm currently implementing #1415, and temporary making new subscripition in main.js.

But now I'm sure adding it to `store`'s would be simpler and much better.

Reply:
> in the event that the package providing the service is deactivated. 

Thanks for the link. The difference is if we are talking about hydrogen deactivating (store Composite disposable gets cleaned up) or the status bar package deactivating ( the disposable returned by the consume function gets cleaned up). I was taking about the former case (@aviadesk was also I believe)

Reply:
It seems very unclear to me what needs a clean up function and what doesn't because I feel like they would have added that to the docs. I asked in the slack and am awaiting a response.

Reply:
It's such a minor thing here I wouldn't worry about it so much. If you're curious try deactivating hydrogen and looking at the dom in devtools after. There will be a Hydrogen status bar element with no size (cant be seen by users). It's been that way forever,but I never thought it was worth looking into.

Reply:
> I was taking about the former case (@aviadesk was also I believe)

My point was exactly what you said there :smile:

Reply:
From Atom
>@Thomas Denton In the example from the Flight Manual, the cleanup that is performed when the consumed package deactivates is all on the side of the consumer. If the consumer deactivates, then it should already know to clean itself up and shouldn't need to use a `Disposable` to do it.

So we shouldn't need to add it to the store. I'll do some testing though

Reply:
![image](https://user-images.githubusercontent.com/40514306/61187676-87d6ce00-a6af-11e9-85dd-ce7b9c984891.png)

Reply:
> If we dispose it when deactivating

![image](https://user-images.githubusercontent.com/40514306/61187797-097b2b80-a6b1-11e9-89ec-f3eb91930026.png)

Reply:
Let's add it to our store.
And note that the disposing an already disposed `Disposable` won't cause the error.

Reply:
Thanks for the testing. Lets add to the store